### PR TITLE
Reset val/test metrics and losses after every epoch

### PR DESCRIPTION
**Pull request by suragnair**
Reset val_losses, val_metrics, test_losses, test_metrics at the end of val/test set so that it does not carry forward to subsequent epochs, as this will not reflect the loss in the latest epoch.

### CODE REVIEW

**Comment by gokceneraslan** [comment]
Cool fix! Any comments on the failed tests?

**Comment by suragnair** [comment]
- seems to happens on CPU and not on GPU
- model training on GPU seems to be normal
- error happens due to `trainer.validate(model=self, dataloaders=val_dataloader)` being called before starting training
- It also has to do with metrics being called in "InferenceMode" in the `trainer.validate` step, and then it is not possible to update them without `torch.inference_mode()` context

**Review by avantikalal — APPROVED** [review]
